### PR TITLE
Hashtags in case management save section dropdowns

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?226215

@emord / @snopoke 

@emord does HQ need to replicate vellum's `isHashtaggable` logic?

### CODE REVIEW

File: corehq/apps/app_manager/xform.py [hunk 1]
```diff
@@ -867,6 +880,9 @@ def get_questions(self, langs, include_triggers=False,
         excluded_paths = set()
 
         control_nodes = self.get_control_nodes()
+        use_hashtags = False
+        if form:
+            use_hashtags = form.get_app().vellum_case_management or toggles.VELLUM_RICH_TEXT.enabled(form.get_app().domain)
```
Comment (dimagimon) [inline corehq/apps/app_manager/xform.py]: E501 line too long (123 > 115 characters)

**Comment by snopoke** [comment]
:+1: so far

**Comment by millerdev** [comment]
👍

**Comment by emord** [comment]
> does HQ need to replicate vellum's isHashtaggable logic?

No I don't think so. That only matters for commtrack and ignore retain, which are advanced stuff anyways

File: corehq/apps/app_manager/xform.py [hunk 2]
```diff
@@ -889,6 +906,8 @@ def get_questions(self, langs, include_triggers=False,
                 "relevant": relevant,
                 "required": required == "true()"
             }
+            if use_hashtags:
+                question.update({ "hashtagValue": self.hashtag_path(path) })
```
Comment (dimagimon) [inline corehq/apps/app_manager/xform.py]: E201 whitespace after '{'
E202 whitespace before '}'

File: corehq/apps/app_manager/xform.py [hunk 3]
```diff
@@ -923,14 +942,22 @@ def get_questions(self, langs, include_triggers=False,
...
                     "value": path,
                     "repeat": matching_repeat_context,
                     "group": matching_repeat_context,
                     "type": "DataBindOnly",
                     "calculate": bind.attrib.get('calculate') if hasattr(bind, 'attrib') else None,
                 }
+                if use_hashtags:
+                    question.update({
+                        "label": hashtag_path,
+                        "hashtagValue": hashtag_path,
```
Comment (dimagimon) [inline corehq/apps/app_manager/xform.py]: F821 undefined name 'hashtag_path'

File: corehq/apps/app_manager/xform.py [hunk 4]
```diff
@@ -923,14 +942,22 @@ def get_questions(self, langs, include_triggers=False,
...
                     "tag": "hidden",
                     "value": path,
                     "repeat": matching_repeat_context,
                     "group": matching_repeat_context,
                     "type": "DataBindOnly",
                     "calculate": bind.attrib.get('calculate') if hasattr(bind, 'attrib') else None,
                 }
+                if use_hashtags:
+                    question.update({
+                        "label": hashtag_path,
```
Comment (dimagimon) [inline corehq/apps/app_manager/xform.py]: F821 undefined name 'hashtag_path'

File: corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js [hunk 5]
```diff
@@ -7,10 +7,12 @@
...
         },
         getDisplay: function (question, MAXLEN) {
-            return utils.getIcon(question) + utils.getLabel(question, MAXLEN) + " (" + question.value + ")";
+            return utils.getIcon(question) + utils.getLabel(question, MAXLEN)
+                    + " (" + (question.hashtagValue || question.value) + ")";
         },
         getLabel: function (question, MAXLEN) {
-            return utils.truncateLabel((question.repeat ? '- ' : '') + question.label, question.tag == 'hidden' ? ' (Hidden)' : '', MAXLEN);
+            return utils.truncateLabel((question.repeat ? '- ' : '')
+                    + question.label, question.tag == 'hidden' ? ' (Hidden)' : '', MAXLEN);
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js]: Expected '===' and instead saw '=='. (eqeqeq)

**Comment by orangejenny** [comment]
Tests pass now. Easiest to review recent commits as one chunk: https://github.com/dimagi/commcare-hq/compare/b78ba9a3cd5c90a087f13ea404f0c118537d77a8...4901f45d8f40c8c7e862d5eb2d30eb4be87baed2

File: corehq/apps/app_manager/xform.py [hunk 6]
```diff
@@ -13,7 +14,9 @@
 from dimagi.utils.decorators.memoized import memoized
 from .xpath import CaseIDXPath, session_var, CaseTypeXpath, QualifiedScheduleFormXPath
 from .exceptions import XFormException, CaseError, XFormValidationError, BindNotFound
+import collections
 import formtranslate.api
+import re
```
Comment (millerdev) [inline corehq/apps/app_manager/xform.py]: nit: std lib modules are normally in the first group of imports (line 3)

**Pull request merged**